frontend/src/views/admin/matching/applicant-view/grid/grid-item/dropdown.tsx

```diff
@@ -35,57 +41,48 @@ export function GridItemDropdown({
     const canBeHidden =
         match.status !== "assigned" && match.status !== "hidden";
 
-    const assignToPosition = React.useCallback(() => {
-        dispatch(
-            upsertMatch({
-                ...baseMatchValues,
-                stagedAssigned: true,
-                stagedHoursAssigned: match.position.hours_per_assignment || 0,
-            })
-        );
-    }, [baseMatchValues, match, dispatch]);
+    const canBeStarred =
+        match.status !== "assigned" && match.status !== "starred";
 
-    const unassignFromPosition = React.useCallback(() => {
-        dispatch(
-            upsertMatch({
-                ...baseMatchValues,
-                stagedAssigned: false,
-                stagedHoursAssigned: 0,
-            })
-        );
-    }, [baseMatchValues, dispatch]);
+    const canBeUnstarred = match.status === "starred";
 
-    const hideFromPosition = React.useCallback(() => {
-        dispatch(upsertMatch({ ...baseMatchValues, hidden: true }));
-    }, [baseMatchValues, dispatch]);
+    const assignApplicant = useAssignApplicantToPosition(
+        baseMatchValues.positionCode,
+        baseMatchValues.utorid,
+        match.position.hours_per_assignment || 0
+    );
+
+    const unassignApplicant = useUnassignApplicantFromPosition(
+        baseMatchValues.positionCode,
+        baseMatchValues.utorid
+    );
+
+    const toggleStar = useToggleStarred(
+        match.position.position_code,
+        match.applicant.utorid
+    );
 
-    const unhideFromPosition = React.useCallback(() => {
-        dispatch(upsertMatch({ ...baseMatchValues, hidden: false }));
-    }, [baseMatchValues, dispatch]);
+    const hideFromAllPositions = useHideFromAllPositions(
+        applicantSummary,
+        true
+    );
 
-    const hideFromAll = React.useCallback(() => {
-        for (const targetMatch of applicantSummary.matches) {
-            dispatch(
-                upsertMatch({
-                    utorid: match.applicant.utorid,
-                    positionCode: targetMatch.position.position_code,
-                    hidden: true,
-                })
-            );
-        }
-    }, [dispatch, match, applicantSummary.matches]);
+    const hideFromPosition = useHideFromPosition(
+        baseMatchValues.positionCode,
+        baseMatchValues.utorid,
+        true
```

Comment:
These extra `true` arguments are very confusing. Why are they needed? Use object literals if you have lots of arguments so that they are self documenting.